internal/error.go

```diff
@@ -196,6 +196,18 @@ type (
 		// VersioningIntent specifies whether the continued workflow should run on a worker with a
 		// compatible build ID or not. See VersioningIntent.
 		VersioningIntent VersioningIntent
+
+		// This is optional and may be used to override the retry policy which gets carried over to the next run.
+		// If not set, the existing retry policy will be carried over automatically.
+		// For backward compatibility, we can't introduce an option, say WithWorkflowRetryPolicy,
+		// to override the retry policy attached to the workflow context.
+		// See #676 for more details.
+		RetryPolicy *commonpb.RetryPolicy
```

Comment:
```suggestion
		// Note, this is always nil when returned from a client as a workflow response.
		RetryPolicy *commonpb.RetryPolicy
```

We don't seem to set this on event so it's not able to be properly set on `workflowRunImpl.GetWithOptions` like a user might expect.

Reply:
It is always `nil` _when returned from a client_ because that is powered by workflow events, and we don't have the retry policy no that event. See `workflowRunImpl.GetWithOptions` to see the logic of how we build this error for clients (not to be confused with how this error is used within workflows).